### DISCUSSION TITLE
Not able to get mesmer metrics

### DISCUSSION THREAD

**Discussion by NandiniMN** [ScalaConsultants/mesmer#544] [Q&A]
We have written simple akka application in scala with a scheduled actor calling another actor for every 5 minutes. We have deployed this application in aws. Currently we are using datadog to check the logs and metrics details. Looking for some help, what we are missing to get the mesmer metrics

Added below dependencies in pom
<dependency>
			<groupId>io.opentelemetry.javaagent</groupId>
			<artifactId>opentelemetry-javaagent</artifactId>
			<version>${opentelemetry-javaagent.version}</version>
		</dependency>
		<dependency>
			<groupId>io.scalac</groupId>
			<artifactId>mesmer-otel-extension_${scala.binary.version}</artifactId>
			<version>${mesmer-akka.version}</version>
		</dependency>
		<dependency>
			<groupId>io.scalac</groupId>
			<artifactId>mesmer-akka-extension_${scala.binary.version}</artifactId>
			<version>${mesmer-akka.version}</version>
		</dependency>
		<dependency>
			<groupId>io.opentelemetry</groupId>
			<artifactId>opentelemetry-exporter-otlp</artifactId>
			<version>1.13.0</version>
		</dependency>
		<dependency>
			<groupId>io.opentelemetry</groupId>
			<artifactId>opentelemetry-sdk-extension-autoconfigure</artifactId>
			<version>1.13.0-alpha</version>
		</dependency>
This the vm args 
      "RuntimeParameters": "-javaagent:/opt/opentelemetry-javaagent-1.18.0.jar -Dotel.exporter.otlp.endpoint=http://127.0.0.1:4317 -Dotel.metric.export.interval=10000 -Dotel.service.name=mesmer-example -Dotel.metrics.exporter=otlp -Dotel.javaagent.extensions=/opt/mesmer-otel-extension_2.13-0.7.0.jar -Xlog:gc*=warning -Dio.scalac.mesmer.module.akkapersistence.recovery.time=false"

And we have this file added
https://github.com/ScalaConsultants/mesmer/blob/main/extension/src/main/resources/reference.conf

**Comment by lgajowy** [comment]
Try to add `-Dotel.javaagent.debug=false` parameter. Do you see the agent/extension starting in the logs right after the start of your application? You should get a whole bunch of (something similar): 

```
[otel.javaagent 2022-11-23 11:02:50:009 +0100] [main] DEBUG io.opentelemetry.javaagent.tooling.AgentInstaller - Loading extension instrumentation-loader [class io.opentelemetry.javaagent.tooling.instrumentation.InstrumentationLoader]
[otel.javaagent 2022-11-23 11:02:50:079 +0100] [main] DEBUG io.opentelemetry.javaagent.tooling.instrumentation.InstrumentationLoader - Loading instrumentation mesmer-akka-http [class io.scalac.mesmer.otelextension.akka.MesmerAkkaHttpInstrumentationModule]
[otel.javaagent 2022-11-23 11:02:50:114 +0100] [main] DEBUG io.opentelemetry.javaagent.tooling.instrumentation.InstrumentationLoader - Loading instrumentation opentelemetry-annotations [class io.opentelemetry.javaagent.instrumentation.otelannotations.WithSpanInstrumentationModule]

```

**Reply by NandiniMN** [reply]
No not getting the extension logs

**Reply by lgajowy** [reply]
This can mean that:

- the javaagent is not being detected (is the path correct)
- the extension is not detected (again, path, but does it point to the [jar from our release](https://github.com/ScalaConsultants/mesmer/releases/tag/v0.7.0)?

Are you able to get other opentelemetry metrics? Not the ones from our extension but the ones the opentelemetry agent provides? If not, that might be the problem with the way you add your agent to your application.

**Comment by NandiniMN** [comment]
we could see only below metrics from datadog
![image](https://user-images.githubusercontent.com/109509107/203526234-e95eb527-9617-4feb-9545-55a06e237d66.png)


And the extension jar version is mesmer-otel-extension_2.13-0.7.0. Do you want me to try with any specific version?
